Core/COEditingContext.h

```diff
@@ -15,6 +15,27 @@
 @class COError;
 
 /**
+ * The behavior to decide when the editing context should unload persistent 
+ * roots.
+ *
+ * TODO: If we keep it around, this should probably become a mask.
+ */
+typedef NS_ENUM(NSUInteger, COEditingContextUnloadingBehavior) {
+	/**
+	 * Persistent roots cannot be unloaded, except uncommitted persistent roots
+	 * on deletion.
+	 */
+	COEditingContextUnloadingBehaviorNever,
+	/**
+	 * Persistent roots can be unloaded on deletion.
+	 *
+     * For external deletions committed in other editing contexts, persistent
+	 * roots will unloaded in the current one.
+	 */
+	COEditingContextUnloadingBehaviorOnDeletion
```

Comment:
I agree.

COEditingContextUnloadingBehaviorNever matters for testing. We can make it private for now.

In future, we can make it public again if we want to expose new options such as COEditingContextUnloadingBehaviorOnMemoryPressure/OnMaxCount. Or if it is helpful for debugging client code.